In [67]:
using LinearAlgebra

Hoje não preciso plotar nenhuma questão, olha que felicidade!

# 1

![grafo](https://i.imgur.com/nMZ4Zez.png)

A primeira coisa que fiz foi passar esses dados da questão para uma matriz. Para encontrar a curva $T(x)= co * x^c1$, me utilizei de uma implementação da matriz de vandermonde onde aplico os valores de x e y pelo inverso da exponencial. Essa questão eu tirei dúvidas com o Thales Macena, para saber como prosseguir, mas busquei realizar minha própria implementação.

In [68]:
xt = [ 0.1 0.2 0.4 0.8 0.9; 
      22 43 84 210 320 ]
auxX = reshape(xt[1,:], 1, length(xt[1,:]))
auxY = reshape(xt[2,:], 1, length(xt[2,:]))
tempLog = zeros(length(auxY),1)
V = zeros(length(auxY),2)

for i=1:length(auxY) #linhas
  tempLog[i,1] = log(auxY[i])
  for j=1:2
    if j==1
      V[i,j]=log(auxX[i])
    else
      V[i,j]=1
    end
  end
end

tempFinal = V\tempLog
tempFinal[2] = ℯ^tempFinal[2]
T(x) = (x^(tempFinal[1]))*tempFinal[2]
t_03 = round(T(0.3), digits=3)

72.061

Assim, obtivemos que $T(0.3) = 72.061$. Ou seja, em 0.3cm de distância, o valor da temperatura  foi de $72.061°C$.

# 2


##### a)

![grafo](https://i.imgur.com/VHaTzNp.png)

Essa questão não tem muito segredo, é aplicação pura de método dos trapézios. Aproximamos uma função por um polinômio de grau 1, fazendo com que ele passe por dois pontos que definem o intervalo de integração. Desta forma, a integral é a área delimitada entre o polinômio e o eixo $x$ no intervalo $a,b$. Passei os dados fornecidos pela questão e apliquei o método.

In [69]:
LaMatrix = [ 0 10 15 20 32 59 62 125;
          0 185 319 447 742 1325 1445 4151]

xMatrix = LaMatrix[1,:]
yMatrix = LaMatrix[2,:]

auxIntegrar = 0
  
for i=1:length(xMatrix)-1
  auxIntegrar = auxIntegrar + ((yMatrix[i+1] + yMatrix[i]) * (xMatrix[i+1] - xMatrix[i]))/2
end

alturaestimada = round(auxIntegrar, digits=3)

219567.5

A altura obtida foi de $219567.5$ pés.

##### b)


![grafo](https://i.imgur.com/7omc1kL.png)

Para que eu estimasse o erro máximo, eu precisaria saber a segunda derivada da função, pois assim saberíamos o valor máximo que ela pode obter. Tendo essa informação, a gente poderia calcular o $M$, consequentemente utilizá-lo na fórmula do erro.

##### c)

As funções foram obtidas do professor, como visto em aula. Não julgo necessário reexplicar método de integração do trapézio ou interpolacão de vandermonde.

In [70]:
function vandermonde(x,y,grau)
  n,=size(y)
  V=zeros(n,grau+1)
  for i=1:n #linhas
      for j=1:(grau+1)
          V[i,j]=x[i]^(j-1)
      end
  end
  return V
end

vandermonde (generic function with 1 method)

In [71]:
function trapezio(f,a,b,n) #calcular a integral f(x) de a até b
  h=(b-a)/n
  S=0.0
  for i=1:(n-1)  #calcula o "meio"
      x=a+i*h
      S+=2*f(x)
  end
  S=h/2*(S+f(a)+f(b)) #calcula "as pontas"
  return S
end

trapezio (generic function with 1 method)

In [72]:
coeficientes = vandermonde(xMatrix,yMatrix,1)\yMatrix
f(x) = sum(coeficientes[i+1]x^i for i in 0:1)
round((trapezio(f, 0, 125, 1000)), digits=2)

224307.61

# 3

##### a)

Nessa questão, eu fiz uma gambiarra para utilizar o método do trapézio sem necessitar de integral dupla: eu transformei a função do círculo na função do arco superior da circunferência. Peguei a função que tínhamos de método do trapézio, modificando o valor de h, já que a questão pediu. Vou chamar a atenção desse carinha, pois ao fixar o tamanho dos trapézios, isso afeta a nossa aproximação.

In [73]:
function trapezio3(f,a,b,n) #calcular a integral f(x) de a até b
  h=0.1
  S=0.0
  for i=1:(n-1)  #calcula o "meio"
      x=a+i*h
      S+=2*f(x)
  end
  S=h/2*(S+f(a)+f(b)) #calcula "as pontas"
  return S
end

trapezio3 (generic function with 1 method)

Tive de criar uma função $funcArcoQuadrantesCima(x)$ para que apenas o domínio faça parte da função e não inclua valores em que a raiz quadrada seja negativa. Algo interessante a se notar é que, por mais que tenha pego uma função do arco superior da circunferência, ao me utilizar da função trapézio, eu tenho de mudar o domínio da integral para pegar apenas o primeiro quadrante.

In [74]:
function funcArcoQuadrantesCima(x)
  if (x >= -1 && x<=1)
    return sqrt(1-(x^2))
  end
  return 0
end

arcoQuadrantesCima(x) = funcArcoQuadrantesCima(x)

intregralQ1 = round((trapezio3(arcoQuadrantesCima, 0, 1, 10000)), digits=2)

0.78

O valor da integral desse quadrante é aproximadamente 0.78, logo precisamos apenas multiplicar esse valor por 4 para obtermos a aproximação de Pi que queremos.

In [75]:
piAprox = intregralQ1 * 4

3.12

Talvez seja estranho não encontrar $3.14$, mas a culpa disso é unicamente por conta da altura que estabelecemos para os trapézios. Veja o que ocorre se utilizamos o método do trapézio sem a restrição da questão:

In [76]:
arco3(x) = primeiroQuadrante(x)

piAprox = round((trapezio(arco3, 0, 1, 10000)*4), digits=2)

3.14

##### b)

Essa é bem simples de explicar, observe o que ocorre ao calcularmos a segunda derivada da função:

$$\left(\sqrt{1-x^2}\right)''\:=-\frac{1}{\left(1-x^2\right)\sqrt{1-x^2}}$$

O erro do trapezio é proporcional à segunda derivada em algum ponto no intervalo de integração. No entanto, a segunda derivada tem uma assintota vertical dentro desse intervalo, logo, não temos como limitar superiormente o erro do método, e isso significa que não podemos afirmar nada sobre ele.

$$-------------$$

Estava com gastroenterite e bastante dor de cabeça esses dias, me poupei de fazer as bônus